Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserComponent.cpp

```diff
@@ -255,17 +255,63 @@ namespace AzToolsFramework
 
                 if (AzFramework::StringFunc::Equal(extension.c_str(), ".ttf"))
                 {
-                    return SourceFileDetails("Icons/AssetBrowser/Font_16.svg");
+                    return SourceFileDetails("Icons/AssetBrowser/Font_80.svg");
                 }
 
                 if (AzFramework::StringFunc::Equal(extension.c_str(), ".xml"))
                 {
-                    return SourceFileDetails("Icons/AssetBrowser/XML_16.svg");
+                    return SourceFileDetails("Icons/AssetBrowser/XML_80.svg");
                 }
 
-                if (AzFramework::StringFunc::Equal(extension.c_str(), ".uicanvas"))
+                if (AzFramework::StringFunc::Equal(extension.c_str(), ".assetinfo"))
                 {
-                    return SourceFileDetails("Icons/AssetBrowser/UICanvas.svg");
+                    return SourceFileDetails("Icons/AssetBrowser/AssetInfo_80.svg");
+                }
+
+                if (AzFramework::StringFunc::Equal(extension.c_str(), ".shader"))
+                {
+                    return SourceFileDetails("Icons/AssetBrowser/Shader_80.svg");
+                }
+
+                if (AzFramework::StringFunc::Equal(extension.c_str(), ".py"))
+                {
+                    return SourceFileDetails("Icons/AssetBrowser/PY_80.svg");
+                }
+
+                if (AzFramework::StringFunc::Equal(extension.c_str(), ".txt"))
+                {
+                    return SourceFileDetails("Icons/AssetBrowser/TXT_80.svg");
+                }
+
+                if (AzFramework::StringFunc::Equal(extension.c_str(), ".json"))
+                {
+                    return SourceFileDetails("Icons/AssetBrowser/JSON_80.svg");
+                }
+
+                if (AzFramework::StringFunc::Equal(extension.c_str(), ".wav"))
+                {
+                    return SourceFileDetails("Icons/AssetBrowser/WAV_80.svg");
+                }
+
+                if (AzFramework::StringFunc::Equal(extension.c_str(), ".wwu"))
+                {
+                    return SourceFileDetails("Icons/AssetBrowser/WWU_80.svg");
+                }
+
+                if (AzFramework::StringFunc::Equal(extension.c_str(), ".wproj"))
+                {
+                    return SourceFileDetails("Icons/AssetBrowser/WPROJ_80.svg");
+                }
+
+                static const char* CFormats[] = { ".c", ".cpp", ".cxx", ".h", ".hpp", ".hxx", ".inl" };
+
+                for (unsigned int CImageFormats = 0, numFormats = AZ_ARRAY_SIZE(CFormats); CImageFormats < numFormats; ++CImageFormats)
+                {
+                    const char* sourceFormatExtension = CFormats[CImageFormats];
+                    if (AzFramework::StringFunc::Equal(extension.c_str(), sourceFormatExtension))
+                    {
+                        return SourceFileDetails("Icons/AssetBrowser/CPP_80");
+                    }
                 }
```

Comment:
We don't show any C or C++ source files in the AssetBrowser, so why do need an icon for CPlusPlus files?

Reply:
The icon was provided by UI/UX, and I tested it by looking at the cpp and header files shown in the AB in the PythonCoverage gem. I don't see any other CPlusPlus files, but I wasn't aware of them being excluded